File: leetCode/array/PascalTriangle_118/Main.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<>();
        List<Integer> layer = new ArrayList<>();
        if(numRows == 0) return triangle;
        layer.add(1);
        triangle.add(layer);
        if(numRows == 1) return triangle;
        List<Integer> prevLayer = layer;
        for (int i = 1; i < numRows; i++) {
            layer = new ArrayList<>();
            layer.add(1);
            for (int j = 1; j < i; j++) {
                int element = prevLayer.get(j - 1) + prevLayer.get(j);
                layer.add(element);
            }
            layer.add(1);
            triangle.add(layer);
            prevLayer = layer;
        }     
        return triangle;
    }
}

public class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();
        var triangle = solution.generate(5);
    }
}

File: leetCode/array/PascalTriangle_118/main.py
class Solution:
    def generate(self, numRows):
        if not numRows:
            return []

        res = [[1]]

        for i in range(1, numRows):
            curr = [1]

            for j in range(1, len(res[i - 1])):
                curr.append(res[i - 1][j - 1] + res[i - 1][j])

            curr.append(1)
            res.append(curr)

        return res